### ISSUE TITLE
User Build - Priority for pattern management in .gitattributes

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/zopeneditor-about#236]
## Development environment used

- Z Open Editor version: 2.0.0
- Editor Platform
  - [X] Visual Studio Code
  - [ ] Red Hat CodeReady Workspaces
  - [ ] Eclipse Che
  - [ ] Standalone Theia
- Editor Platform Version: 1.66.2
- Operating System: Windows 10 20H2
- Java Version: 1.8
- [ ] Related to RSE API?
  - RSE API Plugin version:
  - Zowe CLI version:
  - Node.js version:
- Logs attached: no

## Problem Description

The priority management mode in .gitattributes declarations is not respected.

Z Open Editor stops at the first statement that satisfies the processed file, when it should consider all statements and select the result of merge of all selected patterns, in order of declaration.

See [https://git-scm.com/docs/gitattributes](https://git-scm.com/docs/gitattributes) 

- When more than one pattern matches the path, a later line overrides an earlier line. This overriding is done per attribute.

Consider this use case:

```
# Les fichiers sont gérés et édités en UTF-8 ; sur z/OS, sauf contre indication, ils sont encodés en EBCDIC IBM-1147
*          text=auto git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1147
# debug IBM User Build
*.cbl      git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1047
```
* first pattern say "IBM-1147" for all files
* second pattern say "IBM-1047" for ".cbl" files

```
$ git check-attr --all -- MININAL.cbl
MININAL.cbl: text: auto
MININAL.cbl: git-encoding: utf-8
MININAL.cbl: working-tree-encoding: utf-8
MININAL.cbl: zos-working-tree-encoding: IBM-1047
```

`Uploading file c:\Trv\DepotsGIT\environn-techleads-sandbox\src\S9T\COBOL\MINIMAL.cbl to /u/J070188/vscode/environn-techleads-sandbox/src/S9T/COBOL/MINIMAL.cbl with encoding IBM-1147
`

==> IBM User Build should use IBM-1047

Other use cases:

```
# debug IBM User Build
*.cbl      git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1147
# Les fichiers sont gérés et édités en UTF-8 ; sur z/OS, sauf contre indication, ils sont encodés en EBCDIC IBM-1147
*          text=auto git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1047
```

```
$ git check-attr --all -- MININAL.cbl
MININAL.cbl: text: auto
MININAL.cbl: git-encoding: utf-8
MININAL.cbl: working-tree-encoding: utf-8
MININAL.cbl: zos-working-tree-encoding: IBM-1047
```

`Uploading file c:\Trv\DepotsGIT\environn-techleads-sandbox\src\S9T\COBOL\MINIMAL.cbl to /u/J070188/vscode/environn-techleads-sandbox/src/S9T/COBOL/MINIMAL.cbl with encoding IBM-1147
`

==> IBM User Build should use IBM-1047

```
# debug IBM User Build
*.cbl      git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1047
# Les fichiers sont gérés et édités en UTF-8 ; sur z/OS, sauf contre indication, ils sont encodés en EBCDIC IBM-1147
*          text=auto git-encoding=utf-8 working-tree-encoding=utf-8 zos-working-tree-encoding=IBM-1147
```

```
$ git check-attr --all -- MININAL.cbl
MININAL.cbl: text: auto
MININAL.cbl: git-encoding: utf-8
MININAL.cbl: working-tree-encoding: utf-8
MININAL.cbl: zos-working-tree-encoding: IBM-1147
```

`Uploading file c:\Trv\DepotsGIT\environn-techleads-sandbox\src\S9T\COBOL\MINIMAL.cbl to /u/J070188/vscode/environn-techleads-sandbox/src/S9T/COBOL/MINIMAL.cbl with encoding IBM-1047`

==> IBM User Build should use IBM-1147

**Comment by FALLAI-Denis** [comment]
Hi @phaumer,

We are starting work on setting up IBM DBB. It is imperative for us that the problem described here is solved very quickly. 

Thanks.

**Comment by phaumer** [comment]
Fixed in v2.0.3

**Comment by FALLAI-Denis** [comment]
Thanks @phaumer !

![image](https://user-images.githubusercontent.com/63861965/169594980-51e7daab-eb65-4c0f-a2df-61ce651a74f4.png)
